### PR TITLE
[IndexFilters] Expose prop to control PopoverOverlay

### PR DESCRIPTION
**Pull request by brittcorry**
### WHY are these changes introduced?

There is currently a prop on the `PopoverOverlay` called `preventCloseOnChildOverlayClick` that controls the way the descendants of the overlay are determined. There is also a prop on the `FilterPill` component called `closeonChildOverlayClick` to control this, however it is not exposed any higher to be used by parent components.

In our app (rendered within an iFrame) we're seeing the filter popovers getting stuck open rather than closing when you click another `FilterPill`: [Video Of Problem](https://screenshot.click/24-29-axx2w-9mbdj.mp4)

This problem is solved by passing `true` to the `closeOnChildOverlayClick` prop, therefore exposing that prop to be used by `IndexFilters` is a quick win.

[Video of working filters
](https://screenshot.click/31-04-jvb4w-cthz1.mp4)

### WHAT is this pull request doing?

This PR simply exposes the prop from `FilterPill` higher up to be used by consumers. Nothing in the existing uses of `IndexFilters`, `Filters`, `FilterPill` or `Popover` should be affected.

Testing for the `preventCloseonChildOverlayClick` is already included in the `PopoverOverlay.test.tsx` to ensure its functionality—it was just unusable by consumers of the `Filters` component until now.

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/gold-frogs-sniff.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Exposes closeOnChildOverlayClick prop on IndexFilters to control the preventCloseOnChildOverlayClick prop on PopoverOverlay
```
Comment (chloerice) [inline .changeset/gold-frogs-sniff.md]:
```suggestion
Added support for the `Filters``closeOnChildOverlayClick` prop to `IndexFilters`
```

**Review by chloerice — APPROVED** [review]

**Comment by chloerice** [comment]
Thanks for contributing @brittcorry!! 🙏🏽

**Comment by brittcorry** [comment]
Thanks for the review @chloerice, updated the changeset description! Do I need to merge this PR or just leave it for the team to handle?

**Comment by chloerice** [comment]
No problem @brittcorry, you can merge your PR!

**Pull request merged**